src/v/kafka/server/handlers/topics/topic_utils.h

```diff
@@ -30,32 +30,28 @@
 
 namespace kafka {
 // clang-format off
-CONCEPT(
 template<typename T> 
 concept TopicRequestItem = requires(T item) {
     { item.name } -> std::convertible_to<model::topic_view>;
-};)
-CONCEPT(
+};
 template<typename Iterator> 
 concept TopicResultIterator = requires (Iterator it) {
     it = creatable_topic_result{};
 } && std::is_same_v<typename Iterator::iterator_category, std::output_iterator_tag>;
-)
 // clang-format on
 
 /// Generates failed creatable_topic_result for single topic request item
 template<typename T>
-CONCEPT(requires TopicRequestItem<T>)
-creatable_topic_result
-  generate_error(T item, error_code code, const ss::sstring& msg) {
+requires TopicRequestItem<T> creatable_topic_result
+generate_error(T item, error_code code, const ss::sstring& msg) {
```

Comment:
This is really cool. Now that we don't need to use macro we can do nice things:
```c++
template<TopicRequestItem T>
creatable_topic_result
generate_error(T item, error_code code, const ss::sstring& msg) {
```
Or
```c++
creatable_topic_result generate_error(
  TopicRequestItem auto item, error_code code, const ss::sstring& msg) {
```
...but that's for another time.

Reply:
yeah. also realized that we could use concepts in lots of context!